src/components/mapgl/mapgl.component.ts

```diff
@@ -382,6 +380,17 @@ export class MapglComponent implements OnInit, AfterViewInit, OnChanges {
 
   }
 
+  public emitVisualisations(visu: string) {
+    const visuStatus = this.visualisationsSets.status.get(visu);
+    this.visualisationsSets.status.set(visu, !visuStatus);
+    const layers = new Set<string>();
+    this.visualisationsSets.visualisations.forEach((ls, v) => {
```

Comment:
maybe you could use a filter instead of the `if`?

Reply:
In order to make a future rebase easier, I will treat this in the next PR